--- ocfs2-tools-ocfs2-tools-1.8.8/configure.in.orig	2024-07-22 08:38:57.069461675 +0200
+++ ocfs2-tools-ocfs2-tools-1.8.8/configure.in	2024-07-22 18:13:33.396020729 +0200
@@ -124,6 +124,7 @@ AC_CHECK_HEADER(libaio.h, :,
 AC_SUBST(AIO_LIBS)
 
 NCURSES_LIBS=
+AC_CHECK_LIB(tinfo, tgetstr, NCURSES_LIBS=-ltinfo)
 AC_CHECK_LIB(ncurses, tgetstr, NCURSES_LIBS=-lncurses)
 if test "x$NCURSES_LIBS" = "x"; then
   AC_MSG_ERROR([Unable to find ncurses library])
