--- ocfs2-tools-ocfs2-tools-1.8.6/tunefs.ocfs2/op_query.c~	2019-05-23 12:16:26.000000000 +0200
+++ ocfs2-tools-ocfs2-tools-1.8.6/tunefs.ocfs2/op_query.c	2019-10-20 00:05:43.621595719 +0200
@@ -338,7 +338,7 @@ static int query_run(struct tunefs_opera
 	register_printf_function('O', handle_ro_compat, handle_arginfo);
 
 	query_fs = fs;
-	fprintf(stdout, fmt);
+	fprintf(stdout, "%s", fmt);
 	query_fs = NULL;
 
 	ocfs2_free(&fmt);
